packages/hooks/src/useTransferBetweenAccounts.ts

```diff
@@ -69,79 +71,72 @@ const useTransferBetweenAccounts = () => {
                                 )?.market_type,
                             }),
                         };
-                    }) || [],
+
+                        return all_accounts;
+                    },
+                    {} as Record<
+                        string,
+                        NonNullable<
+                            typeof accounts[number] & {
+                                gradient_class?: `wallet-card__${string}`;
+                                icon?: string;
+                                mt5_market_type?: 'all' | 'financial' | 'synthetic';
+                            }
+                        >
+                    >
+                ) || {},
             wallets:
-                accounts
-                    ?.filter(account => account.account_type === 'wallet')
-                    .map(wallet_account => {
-                        const wallet = all_wallets?.find(acc => acc.loginid === wallet_account.loginid);
-
-                        return {
-                            ...wallet_account,
-                            icon: wallet?.icon,
-                            gradient_class: wallet?.gradient_card_class,
+                accounts?.reduce(
+                    (all_wallets, wallet) => {
+                        if (wallet.account_type !== 'wallet') return all_wallets;
+                        if (!wallet.loginid) return all_wallets;
+
+                        const available_wallet = wallets?.find(acc => acc.loginid === wallet.loginid);
+
+                        all_wallets[wallet.loginid] = {
+                            ...wallet,
+                            icon: available_wallet?.icon,
+                            gradient_class: available_wallet?.gradient_card_class,
                         };
-                    }) || [],
+
+                        return all_wallets;
+                    },
+                    {} as Record<
+                        string,
+                        NonNullable<
+                            typeof accounts[number] & {
+                                gradient_class?: `wallet-card__${string}`;
+                                icon?: string;
+                            }
+                        >
+                    >
```

Comment:
Move these type definition to it's own type declaration and reuse it with the above